src/java/htsjdk/samtools/fastq/FastqRecord.java

```diff
@@ -24,18 +24,23 @@
 package htsjdk.samtools.fastq;
 
 import java.io.Serializable;
+import java.util.Objects;
 
 /**
  * Represents a fastq record, fairly literally, i.e. without any conversion.
  */
 public class FastqRecord implements Serializable {
     private static final long serialVersionUID = 1L;
     private final String seqHeaderPrefix;
-    private final String seqLine;
+    private final String seqLine;            //not null, not empty
```

Comment:
Just commenting to note that, while I think this is good and the right thing to do, this is a breaking change that will effect many places in the ecosystem, and so should be clearly noted in release notes and a corresponding bump in version number made.